lib/rules/array-element-newline.js

```diff
@@ -0,0 +1,250 @@
+        * @param {Token} token - The token to use for the report.
+        * @returns {void}
+        */
+        function reportRequiredLineBreak(node, token) {
+            const tokenBefore = sourceCode.getTokenBefore(token, { includeComments: true });
+
+            context.report({
+                node,
+                loc: {
+                    start: tokenBefore.loc.end,
+                    end: token.loc.start
+                },
+                message: "Here should be a linebreak.",
+                fix(fixer) {
+                    return fixer.replaceTextRange([tokenBefore.range[1], token.range[0]], "\n");
+                }
+            });
+        }
+
+        /**
+         * Reports a given node if it violated this rule.
+         *
+         * @param {ASTNode} node - A node to check. This is an ObjectExpression node or an ObjectPattern node.
+         * @param {{multiline: boolean, minItems: number}} options - An option object.
+         * @returns {void}
+         */
+        function check(node) {
+            const elements = node.elements;
+            const normalizedOptions = normalizeOptions(context.options[0]);
+            const options = normalizedOptions[node.type];
+
+            let elementBreak = false;
+
+            /*
+             * MULTILINE: true
+             * loop through every element and check
+             * if at least one element has linebreaks inside
+             * this ensures that following is not valid (due to elements are on the same line):
+             *
+             * [
+             *      1,
+             *      2,
+             *      3
+             * ]
+             */
+            if (options.multiline) {
+                for (let i = 0; i < elements.length; i++) {
+                    const element = elements[i];
+
+                    if (element === null) {
+                        continue;
+                    }
+
+                    const elementFirst = sourceCode.getFirstToken(element);
+                    const elementLast = sourceCode.getLastToken(element);
+
+                    if (options.multiline && !astUtils.isTokenOnSameLine(elementFirst, elementLast)) {
+                        elementBreak = true;
+
+                        break;
+                    }
+                }
+            }
+
+            const needsLinebreaks = (
+                elements.length >= options.minItems ||
+                (
+                    options.multiline &&
+                    elements.length > 0 &&
+                    elementBreak
+                )
+            );
+
+            elements.forEach((element, i) => {
+                if (i === 0 || element === null) {
+                    return;
+                }
+
+                const lastTokenOfPreviousElement = sourceCode.getTokenBefore(element, { skip: 1 });
+                const firstTokenOfCurrentElement = sourceCode.getFirstToken(element);
```

Comment:
This will work incorrectly if an element has parentheses:

```js
// this should be valid with "always", but the rule reports an error

[
    a
    , (b)
]
```